reload.go

```diff
@@ -58,7 +59,16 @@ func Do(log func(string, ...interface{})) error {
 				// Panic, and Fatal :-/ Printf() is probably best.
 				log("reload error: %v", err)
 			case event := <-watcher.Events:
-				if event.Op&fsnotify.Write == fsnotify.Write && event.Name == bin {
+				var eventTriggered bool
+
+				// workaround for macOS, which does not trigger WRITEs
+				if runtime.GOOS == "darwin" {
+					eventTriggered = event.Op&fsnotify.Create == fsnotify.Create
+				} else {
+					eventTriggered = event.Op&fsnotify.Write == fsnotify.Write
+				}
```

Comment:
Could you expand this to `freebsd`, `openbsd`, `netbsd`, and `dragonfly` as well? They all use the same kqueue, so I would *expect* them to behave the same. Using a `switch` would probably be clearer.

Might also be a good idea to warn on untested GOOS like so:

```
// Ensure that we use the correct events, as they are not uniform accross
// platforms. See https://github.com/fsnotify/fsnotify/issues/74
var trigger bool
switch runtime.GOOS {
case "darwin", "openbsd", [..]:
    trigger = event.Op&fsnotify.Create == fsnotify.Create
case "linux":
    trigger = event.Op&fsnotify.Write== fsnotify.Write
default:
    trigger = event.Op&fsnotify.Create == fsnotify.Create
    log("reload: untested GOOS %q; this package may not work correctly", runtime.GOOS)
}
```

Reply:
Sure